app crash on CourseStepFragment.onDestroy with Realm NullPointer Exception

```
FATAL EXCEPTION: main
Process: org.ole.planet.myplanet, PID: 8579
java.lang.RuntimeException: Unable to destroy activity {org.ole.planet.myplanet/org.ole.planet.myplanet.ui.dashboard.
DashboardActivity}: java.lang.
NullPointerException: Attempt to invoke virtual method 'void io.realm.Realm.close()' on a null object reference
at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5126)
at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:5155)
at android.app.servertransaction.DestroyActivityItem.execute(DestroyActivityItem.java:44)
at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2077)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loop(Looper.java:223)
at android.app.ActivityThread.main(ActivityThread.java:7741)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:966)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void io.realm.Realm.close()' on a null object reference
at org.ole.planet.myplanet.ui.course.CourseStepFragment.onDestroy(CourseStepFragment.java:117)
at androidx.fragment.app.Fragment.performDestroy(Fragment.java:3360)
at androidx.fragment.app.FragmentStateManager.destroy(FragmentStateManager.java:781)
at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:335)
at androidx.fragment.app.FragmentStore.moveToExpectedState(FragmentStore.java:113)
at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1424)
at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2968)
at androidx.fragment.app.FragmentManager.dispatchDestroy(FragmentManager.java:2922)
at androidx.fragment.app.Fragment.performDestroy(Fragment.java:3355)
at androidx.fragment.app.FragmentStateManager.destroy(FragmentStateManager.java:781)
at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:335)
at androidx.fragment.app.FragmentStore.moveToExpectedState(FragmentStore.java:121)
at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1424)
at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2968)
at androidx.fragment.app.FragmentManager.dispatchDestroy(FragmentManager.java:2922)
at androidx.fragment.app.FragmentController.dispatchDestroy(FragmentController.java:346)
at androidx.fragment.app.FragmentActivity.onDestroy(FragmentActivity.java:259)
at androidx.appcompat.app.AppCompatActivity.onDestroy(AppCompatActivity.java:283)
at org.ole.planet.myplanet.ui.dashboard.DashboardActivity.onDestroy(DashboardActivity.java:335)
at android.app.Activity.performDestroy(Activity.java:8255)
at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1362)
at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5111)
at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:5155) 
at android.app.servertransaction.DestroyActivityItem.execute(DestroyActivityItem.java:44) 
at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176) 
at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97) 
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2077) 
at android.os.Handler.dispatchMessage(Handler.java:106) 
at android.os.Looper.loop(Looper.java:223) 
at android.app.ActivityThread.main(ActivityThread.java:7741) 
at java.lang.reflect.Method.invoke(Native Method) 
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592) 
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:966) 
```

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -10,8 +10,8 @@ android {
         applicationId "org.ole.planet.myplanet"
         minSdkVersion 21
         targetSdkVersion 34
-        versionCode 1106
-        versionName "0.11.6"
+        versionCode 1107
+        versionName "0.11.7"
         ndkVersion '21.3.6528147'
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
         vectorDrawables.useSupportLibrary = true

--- a/app/src/main/java/org/ole/planet/myplanet/ui/course/CourseStepFragment.java
+++ b/app/src/main/java/org/ole/planet/myplanet/ui/course/CourseStepFragment.java
@@ -102,7 +102,9 @@ public class CourseStepFragment extends BaseContainerFragment implements CameraU
     @Override
     public void onDestroy() {
         super.onDestroy();
-        mRealm.close();
+        if (mRealm != null) {
+            mRealm.close();
+        }
     }
 
     @Override

